network/p2p/middleware.go

```diff
@@ -332,7 +333,7 @@ func (m *Middleware) handleIncomingStream(s libp2pnetwork.Stream) {
 	log.Info().Msg("incoming connection established")
 
 	//create a new readConnection with the context of the middleware
-	conn := newReadConnection(m.ctx, s, m.processMessage, log, m.metrics, m.maxUnicastMsgSize)
+	conn := newReadConnection(m.ctx, s, m.processMessage, log, m.metrics, LargeMsgMaxUnicastMsgSize)
```

Comment:
The `readConnection` has to expect a message of upto `LargeMsgMaxUnicastMsgSize`. Once the message is received, its type can be used to determine if it exceeds size but that can only happen after it has been _fully_ received.
So, this does unfortunately introduce the possibility that a bad sender can keep sending large messages (other than `messages.ChunkDataResponse`) of size > 2mb and unnecessarily keep the received busy doing i/o.

Reply:
Thanks for pointing this, I think it is worthy of opening an issue as part of _benign byzantine_ vulnerabilities of network layer.

Reply:
https://github.com/dapperlabs/flow-go/issues/5191